src/main/java/hudson/plugins/git/GitPublisher.java

```diff
@@ -185,7 +186,7 @@ public boolean perform(AbstractBuild<?, ?> build,
         SCM scm = build.getProject().getScm();
 
         if (!(scm instanceof GitSCM)) {
-            return false;
+            return true; // just skip this publisher
```

Comment:
Right in my plugin i finally found a solution by using f.property() with this class https://github.com/KostyaSha/github-pullrequest-plugin/blob/master/src/main/java/org/jenkinsci/plugins/github/pullrequest/utils/PublisherErrorHandler.java
In terms of this PR reverting to false will be right.